torchvision/models/swin_transformer.py

```diff
@@ -304,6 +457,57 @@ def forward(self, x: Tensor):
         return x
 
 
+class SwinTransformerBlockV2(SwinTransformerBlock):
+    """
+    Swin Transformer V2 Block.
+    Args:
+        dim (int): Number of input channels.
+        num_heads (int): Number of attention heads.
+        window_size (List[int]): Window size.
+        shift_size (List[int]): Shift size for shifted window attention.
+        mlp_ratio (float): Ratio of mlp hidden dim to embedding dim. Default: 4.0.
+        dropout (float): Dropout rate. Default: 0.0.
+        attention_dropout (float): Attention dropout rate. Default: 0.0.
+        stochastic_depth_prob: (float): Stochastic depth rate. Default: 0.0.
+        norm_layer (nn.Module): Normalization layer.  Default: nn.LayerNorm.
+        attn_layer (nn.Module): Attention layer. Default: ShiftedWindowAttentionV2.
+        pretrained_window_size (int): Local window size in pre-training. Default: 0.
+    """
+
+    def __init__(
+        self,
+        dim: int,
+        num_heads: int,
+        window_size: List[int],
+        shift_size: List[int],
+        mlp_ratio: float = 4.0,
+        dropout: float = 0.0,
+        attention_dropout: float = 0.0,
+        stochastic_depth_prob: float = 0.0,
+        norm_layer: Callable[..., nn.Module] = nn.LayerNorm,
+        attn_layer: Callable[..., nn.Module] = ShiftedWindowAttentionV2,
+        pretrained_window_size: int = 0,
```

Comment:
Does this pretrained_window_size always has the same value for all the dimension? Just for context, we may expand similar API to Swin Transformer Video, and usually for video the window_size for time dimension is different with height and width, therefore we use `List[int]` at the end.

Reply:
It's only used for some edge cases: That means The model is pretrained with `pretrained_window size`, but uses a different `window_size` at evaluation time.

You can view those cases in the yamls:
https://github.com/search?q=pretrained_window_sizes+repo%3Amicrosoft%2FSwin-Transformer+filename%3A*.yaml&type=Code&ref=advsearch&l=&l=